File: app/src/main/java/com/veganbeauty/app/features/myskin/BookingAdapters.java
package com.veganbeauty.app.features.myskin;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.R;

import java.util.Calendar;
import java.util.List;

public class BookingAdapters {

    public static class BookingService {
        private final String id;
        private final String name;
        private final String description;
        private final String duration;

        public BookingService(String id, String name, String description, String duration) {
            this.id = id;
            this.name = name;
            this.description = description;
            this.duration = duration;
        }

        public String getId() { return id; }
        public String getName() { return name; }
        public String getDescription() { return description; }
        public String getDuration() { return duration; }
    }

    public static class BookingDate {
        private final String id;
        private final String dayOfWeek;
        private final String date;
        private final Calendar fullDate;

        public BookingDate(String id, String dayOfWeek, String date, Calendar fullDate) {
            this.id = id;
            this.dayOfWeek = dayOfWeek;
            this.date = date;
            this.fullDate = fullDate;
        }

        public String getId() { return id; }
        public String getDayOfWeek() { return dayOfWeek; }
        public String getDate() { return date; }
        public Calendar getFullDate() { return fullDate; }
    }

    public static class BookingTime {
        private final String id;
        private final String time;
        private final boolean isLocked;

        public BookingTime(String id, String time, boolean isLocked) {
            this.id = id;
            this.time = time;
            this.isLocked = isLocked;
        }

        public String getId() { return id; }
        public String getTime() { return time; }
        public boolean isLocked() { return isLocked; }
    }

    public interface OnServiceSelectedListener {
        void onSelected(BookingService service);
    }

    public interface OnDateSelectedListener {
        void onSelected(BookingDate date);
    }

    public interface OnTimeSelectedListener {
        void onSelected(BookingTime time);
    }

    public static class BookingServiceAdapter extends RecyclerView.Adapter<BookingServiceAdapter.ViewHolder> {
        private final List<BookingService> items;
        private final OnServiceSelectedListener onItemSelected;
        private int selectedIndex = -1;

        public BookingServiceAdapter(List<BookingService> items, OnServiceSelectedListener onItemSelected) {
            this.items = items;
            this.onItemSelected = onItemSelected;
        }

        public static class ViewHolder extends RecyclerView.ViewHolder {
            public final View container;
            public final TextView tvName;
            public final TextView tvDesc;
            public final ImageView ivCheck;

            public ViewHolder(View view) {
                super(view);
                container = view.findViewById(R.id.container_service);
                tvName = view.findViewById(R.id.tv_service_name);
                tvDesc = view.findViewById(R.id.tv_service_desc);
                ivCheck = view.findViewById(R.id.iv_service_check);
            }
        }

        @NonNull
        @Override
        public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.skin_item_booking_service, parent, false);
            return new ViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
            BookingService item = items.get(position);
            holder.tvName.setText(item.getName());
            holder.tvDesc.setText(item.getDescription());

            boolean isSelected = position == selectedIndex;
            holder.ivCheck.setVisibility(isSelected ? View.VISIBLE : View.GONE);

            if (isSelected) {
                holder.container.setBackgroundResource(R.drawable.skin_bg_selected_item);
            } else {
                holder.container.setBackgroundResource(R.drawable.skin_bg_outline);
            }

            holder.container.setOnClickListener(v -> {
                int oldIndex = selectedIndex;
                selectedIndex = holder.getAdapterPosition();
                notifyItemChanged(oldIndex);
                notifyItemChanged(selectedIndex);
                if (onItemSelected != null) {
                    onItemSelected.onSelected(item);
                }
            });
        }

        @Override
        public int getItemCount() {
            return items.size();
        }
    }

    public static class BookingDateAdapter extends RecyclerView.Adapter<BookingDateAdapter.ViewHolder> {
        private final List<BookingDate> items;
        private final OnDateSelectedListener onItemSelected;
        private int selectedIndex = -1;

        public BookingDateAdapter(List<BookingDate> items, OnDateSelectedListener onItemSelected) {
            this.items = items;
            this.onItemSelected = onItemSelected;
        }

        public static class ViewHolder extends RecyclerView.ViewHolder {
            public final View container;
            public final TextView tvDay;
            public final TextView tvDate;

            public ViewHolder(View view) {
                super(view);
                container = view.findViewById(R.id.container_date);
                tvDay = view.findViewById(R.id.tv_date_day);
                tvDate = view.findViewById(R.id.tv_date_date);
            }
        }

        @NonNull
        @Override
        public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.skin_item_booking_date, parent, false);
            return new ViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
            BookingDate item = items.get(position);
            holder.tvDay.setText(item.getDayOfWeek());
            holder.tvDate.setText(item.getDate());

            boolean isSelected = position == selectedIndex;
            if (isSelected) {
                holder.container.setBackgroundResource(R.drawable.skin_bg_selected_item);
            } else {
                holder.container.setBackgroundResource(R.drawable.skin_bg_outline);
            }

            holder.container.setOnClickListener(v -> {
                int oldIndex = selectedIndex;
                selectedIndex = holder.getAdapterPosition();
                notifyItemChanged(oldIndex);
                notifyItemChanged(selectedIndex);
                if (onItemSelected != null) {
                    onItemSelected.onSelected(item);
                }
            });
        }

        @Override
        public int getItemCount() {
            return items.size();
        }

        public void setSelectedIndex(int index) {
            int oldIndex = selectedIndex;
            selectedIndex = index;
            notifyItemChanged(oldIndex);
            notifyItemChanged(selectedIndex);
        }
    }

    public static class BookingTimeAdapter extends RecyclerView.Adapter<BookingTimeAdapter.ViewHolder> {
        private List<BookingTime> items;
        private final OnTimeSelectedListener onItemSelected;
        private int selectedIndex = -1;

        public BookingTimeAdapter(List<BookingTime> items, OnTimeSelectedListener onItemSelected) {
            this.items = items;
            this.onItemSelected = onItemSelected;
        }

        public static class ViewHolder extends RecyclerView.ViewHolder {
            public final View container;
            public final TextView tvTime;

            public ViewHolder(View view) {
                super(view);
                container = view.findViewById(R.id.container_time);
                tvTime = view.findViewById(R.id.tv_time);
            }
        }

        @NonNull
        @Override
        public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.skin_item_booking_time, parent, false);
            return new ViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
            BookingTime item = items.get(position);
            holder.tvTime.setText(item.getTime());

            boolean isSelected = position == selectedIndex;

            if (item.isLocked()) {
                holder.container.setBackgroundResource(R.drawable.skin_bg_outline);
                holder.container.setAlpha(0.4f);
                holder.tvTime.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.content));
            } else {
                holder.container.setAlpha(1.0f);
                if (isSelected) {
                    holder.container.setBackgroundResource(R.drawable.skin_bg_selected_item);
                    holder.tvTime.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.primary));
                } else {
                    holder.container.setBackgroundResource(R.drawable.skin_bg_outline);
                    holder.tvTime.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.content));
                }
            }

            holder.container.setOnClickListener(v -> {
                if (item.isLocked()) return;
                
                int oldIndex = selectedIndex;
                selectedIndex = holder.getAdapterPosition();
                notifyItemChanged(oldIndex);
                notifyItemChanged(selectedIndex);
                if (onItemSelected != null) {
                    onItemSelected.onSelected(item);
                }
            });
        }

        @Override
        public int getItemCount() {
            return items.size();
        }

        public void updateData(List<BookingTime> newItems) {
            items = newItems;
            selectedIndex = -1;
            notifyDataSetChanged();
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/myskin/BookingAdapters.kt
package com.veganbeauty.app.features.myskin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.veganbeauty.app.R

// --- Data Classes ---
data class BookingService(val id: String, val name: String, val description: String, val duration: String)
data class BookingDate(val id: String, val dayOfWeek: String, val date: String, val fullDate: java.util.Calendar? = null)
data class BookingTime(val id: String, val time: String, val isLocked: Boolean = false)

// --- BookingServiceAdapter ---
class BookingServiceAdapter(
    private val items: List<BookingService>,
    private val onItemSelected: (BookingService) -> Unit
) : RecyclerView.Adapter<BookingServiceAdapter.ViewHolder>() {

    private var selectedIndex = -1

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val container: View = view.findViewById(R.id.container_service)
        val tvName: TextView = view.findViewById(R.id.tv_service_name)
        val tvDesc: TextView = view.findViewById(R.id.tv_service_desc)
        val ivCheck: ImageView = view.findViewById(R.id.iv_service_check)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.skin_item_booking_service, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        holder.tvName.text = item.name
        holder.tvDesc.text = item.description

        val isSelected = position == selectedIndex
        holder.ivCheck.visibility = if (isSelected) View.VISIBLE else View.GONE
        
        if (isSelected) {
            holder.container.setBackgroundResource(R.drawable.skin_bg_selected_item)
        } else {
            holder.container.setBackgroundResource(R.drawable.skin_bg_outline)
        }

        holder.container.setOnClickListener {
            val oldIndex = selectedIndex
            selectedIndex = position
            notifyItemChanged(oldIndex)
            notifyItemChanged(selectedIndex)
            onItemSelected(item)
        }
    }

    override fun getItemCount(): Int = items.size
}

// --- BookingDateAdapter ---
class BookingDateAdapter(
    private val items: List<BookingDate>,
    private val onItemSelected: (BookingDate) -> Unit
) : RecyclerView.Adapter<BookingDateAdapter.ViewHolder>() {

    private var selectedIndex = -1

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val container: View = view.findViewById(R.id.container_date)
        val tvDay: TextView = view.findViewById(R.id.tv_date_day)
        val tvDate: TextView = view.findViewById(R.id.tv_date_date)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.skin_item_booking_date, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        holder.tvDay.text = item.dayOfWeek
        holder.tvDate.text = item.date

        val isSelected = position == selectedIndex
        if (isSelected) {
            holder.container.setBackgroundResource(R.drawable.skin_bg_selected_item)
        } else {
            holder.container.setBackgroundResource(R.drawable.skin_bg_outline)
        }

        holder.container.setOnClickListener {
            val oldIndex = selectedIndex
            selectedIndex = position
            notifyItemChanged(oldIndex)
            notifyItemChanged(selectedIndex)
            onItemSelected(item)
        }
    }

    override fun getItemCount(): Int = items.size
    
    fun setSelectedIndex(index: Int) {
        val oldIndex = selectedIndex
        selectedIndex = index
        notifyItemChanged(oldIndex)
        notifyItemChanged(selectedIndex)
    }
}

// --- BookingTimeAdapter ---
class BookingTimeAdapter(
    private var items: List<BookingTime>,
    private val onItemSelected: (BookingTime) -> Unit
) : RecyclerView.Adapter<BookingTimeAdapter.ViewHolder>() {

    private var selectedIndex = -1

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val container: View = view.findViewById(R.id.container_time)
        val tvTime: TextView = view.findViewById(R.id.tv_time)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.skin_item_booking_time, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        holder.tvTime.text = item.time

        val isSelected = position == selectedIndex
        
        if (item.isLocked) {
            holder.container.setBackgroundResource(R.drawable.skin_bg_outline)
            holder.container.alpha = 0.4f
            holder.tvTime.setTextColor(ContextCompat.getColor(holder.itemView.context, R.color.content))
        } else {
            holder.container.alpha = 1.0f
            if (isSelected) {
                holder.container.setBackgroundResource(R.drawable.skin_bg_selected_item)
                holder.tvTime.setTextColor(ContextCompat.getColor(holder.itemView.context, R.color.primary))
            } else {
                holder.container.setBackgroundResource(R.drawable.skin_bg_outline)
                holder.tvTime.setTextColor(ContextCompat.getColor(holder.itemView.context, R.color.content))
            }
        }

        holder.container.setOnClickListener {
            if (item.isLocked) return@setOnClickListener
            
            val oldIndex = selectedIndex
            selectedIndex = position
            notifyItemChanged(oldIndex)
            notifyItemChanged(selectedIndex)
            onItemSelected(item)
        }
    }

    override fun getItemCount(): Int = items.size
    
    fun updateData(newItems: List<BookingTime>) {
        items = newItems
        selectedIndex = -1
        notifyDataSetChanged()
    }
}
